File: MVCProject/app/src/main/java/com/example/mvc_project/db/AppDataBase.java
package com.example.mvc_project.db;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.mvc_project.model.Product;

@Database(entities = {Product.class}, version = 1 )
public abstract class AppDataBase extends RoomDatabase {
    private static AppDataBase instance = null;
    public abstract ProductDAO productDAO();

    public static synchronized AppDataBase getInstance( Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(), AppDataBase.class, "productdb").build();
        }
        return instance;
    }
}

File: MVVM/app/src/main/java/com/example/mvvm/database/AppDatabase.kt
package com.example.mvvm.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.mvvm.model.Product

@Database(entities = arrayOf(Product::class), version = 3)
abstract class AppDatabase :RoomDatabase() {
    abstract fun getProductDao() : ProductDao
    companion object {
        @Volatile
        private var INSTANCE :AppDatabase?=null

        fun getInstance (ctx: Context):AppDatabase{
            return INSTANCE?: synchronized(this){
                val instance  = Room.databaseBuilder(
                    ctx.applicationContext,AppDatabase::class.java,"Products_DataBase3"
                ).build()
                INSTANCE = instance
                instance
            }
        }

    }
}